src/sagemaker/lineage/action.py

```diff
@@ -250,3 +259,107 @@ def list(
             max_results=max_results,
             next_token=next_token,
         )
+
+    def upstream_artifacts(
+        self, direction: LineageQueryDirectionEnum = LineageQueryDirectionEnum.ASCENDANTS
```

Comment:
If we're going the route of having a separate upstream/downstream method for each query, we shouldn't allow the direction to be changed. 

I'm wondering if it's worth just having an `artifacts` method with direction defaulted to BOTH. Users can change to ASCENDANTS/DESCENDANTS based on their needs

Reply:
Either way, let's try to cut down on the code duplication if possible. Have a `_artifacts()` method that both of these call with the common code

Reply:
> Either way, let's try to cut down on the code duplication if possible. Have a `_artifacts()` method that both of these call with the common code

To make it consistent with other methods, I updated to have `artifacts` with default both direction. The user can update the direction to upstream, downstream or both if they want to